Updates to computer vision section of the Preprocess doc (#21181)
huggingface/transformers: docs/source/en/preprocessing.mdx

1. Here we use [`Compose`](https://pytorch.org/vision/master/generated/torchvision.transforms.Compose.html) to chain together a couple of
transforms - [`RandomResizedCrop`](https://pytorch.org/vision/main/generated/torchvision.transforms.RandomResizedCrop.html) and [`ColorJitter`](https://pytorch.org/vision/main/generated/torchvision.transforms.ColorJitter.html).
Note that for resizing, we can get the image size requirements from the `image_processor`. For some models, an exact height and
width are expected, for others only the `shortest_edge` is defined.

```py
>>> from torchvision.transforms import RandomResizedCrop, ColorJitter, Compose

>>> size = (
...     image_processor.size["shortest_edge"]
...     if "shortest_edge" in image_processor.size
...     else (image_processor.size["height"], image_processor.size["width"])
... )

>>> _transforms = Compose([RandomResizedCrop(size), ColorJitter(brightness=0.5, hue=0.5)])
```

2. The model accepts [`pixel_values`](model_doc/visionencoderdecoder#transformers.VisionEncoderDecoderModel.forward.pixel_values)

> as its input. ImageProcessor can take care of normalizing the images, and generating appropriate tensors.

Reviewer: ```suggestion
as its input. `ImageProcessor` can take care of normalizing the images, and generating appropriate tensors.
```

Create a function that combines image augmentation and image preprocessing for a batch of images and generates `pixel_values`:

```py
>>> def transforms(examples):
...     images = [_transforms(img.convert("RGB")) for img in examples["image"]]
...     examples["pixel_values"] = image_processor(images, do_resize=False, return_tensors="pt")["pixel_values"]
...     return examples
```

<Tip>

In the example above we set `do_resize=False` because we have already resized the images in the image augmentation transformation,
and leveraged the `size` attribute from the appropriate `image_processor`. If you do not resize images during image augmentation,

> leave this parameter out. By default, ImageProcessor will handle the resizing.

Reviewer: ```suggestion
leave this parameter out. By default, `ImageProcessor` will handle the resizing.
```

    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/preprocessed_image.png"/>
</div>

<Tip>

> For tasks like object detection, semantic segmentation, instance segmentation, and panoptic segmentation, ImageProcessor

Reviewer: ```suggestion
For tasks like object detection, semantic segmentation, instance segmentation, and panoptic segmentation, `ImageProcessor`
```

offers post processing methods. These methods convert model's raw outputs into meaningful predictions such as bounding boxes,
or segmentation maps.

</Tip>

> ### Pad

Reviewer: Add a blank line after the section header to correctly render the links below :)

Author: Thanks! I need to get used to the blank lines after headers. Fixed in 2f30d954b314406746b125b5be97cca894a97d17

In some cases, for instance, when fine-tuning [DETR](./model_doc/detr), the model applies scale augmentation at training

> time. This may cause images of different sizes in a batch. You can use [`DetrImageProcessor.pad_and_create_pixel_mask`]

Reviewer: ```suggestion
time. This may cause images to be different sizes in a batch. You can use [`DetrImageProcessor.pad_and_create_pixel_mask`]
```
